test/proxy/mzcompose.py

```diff
@@ -7,78 +7,93 @@
+from typing import Dict, List
 
-from materialize.mzcompose import Composition
+from materialize.mzcompose import Composition, WorkflowArgumentParser
 from materialize.mzcompose.services import (
     Kafka,
     Localstack,
     Materialized,
+    Redpanda,
     SchemaRegistry,
     Squid,
     Testdrive,
     Zookeeper,
 )
 
-prerequisites = ["zookeeper", "kafka", "schema-registry", "squid", "localstack"]
-
-# Run certain testdrive tests for each combination of env variables under test
-tests: List[Dict[str, Any]] = [
-    {
-        "name": "with_proxy",
-        "env": ["ALL_PROXY=http://squid:3128"],
-        "td": ["testdrive/avro-registry.td", "testdrive/esoteric/s3.td"],
-    },
-    {
-        "name": "proxy_failure",
-        "env": ["ALL_PROXY=http://localhost:1234"],
-        "td": ["proxy-failure.td"],
-    },
-    {
-        "name": "no_proxy",
-        "env": [
-            "ALL_PROXY=http://localhost:1234",
-            "NO_PROXY=schema-registry,amazonaws.com,localstack",
-        ],
-        "td": ["testdrive/avro-registry.td", "testdrive/esoteric/s3.td"],
-    },
-]
-
-# Construct a dedicated Mz instance for each set of env variables under test
-for t in tests:
-    t["mz"] = Materialized(
-        name=f"materialized_{t['name']}",
-        hostname="materialized",
-        environment_extra=t["env"],
-    )
-
-mzs = [t["mz"] for t in tests]
-
 SERVICES = [
     Zookeeper(),
     Kafka(),
     SchemaRegistry(),
+    Redpanda(),
     Squid(),
     Localstack(),
-    *mzs,
+    Materialized(),
     Testdrive(volumes_extra=["../testdrive:/workdir/testdrive"]),
 ]
 
 
-def workflow_proxy(c: Composition) -> None:
-    c.start_and_wait_for_tcp(services=prerequisites)
-    test_proxy(c, "--aws-endpoint=http://localstack:4566")
+@dataclass
+class TestCase:
+    name: str
+    env: List[str]
+    files: List[str]
 
 
-def workflow_proxy_ci(c: Composition) -> None:
-    c.start_and_wait_for_tcp(services=prerequisites)
-    test_proxy(c, "--aws-region=us-east-2")
+test_cases = [
+    TestCase(
+        name="with-proxy",
+        env=["ALL_PROXY=http://squid:3128"],
+        files=["testdrive/avro-registry.td", "testdrive/esoteric/s3.td"],
+    ),
+    TestCase(
+        name="proxy-failure",
+        env=["ALL_PROXY=http://squid:3128"],
+        files=["testdrive/avro-registry.td", "testdrive/esoteric/s3.td"],
+    ),
+    TestCase(
+        name="no-proxy",
+        env=[
+            "ALL_PROXY=http://localhost:1234",
+            "NO_PROXY=schema-registry,amazonaws.com,localstack",
+        ],
+        files=["testdrive/avro-registry.td", "testdrive/esoteric/s3.td"],
+    ),
+]
 
 
-def test_proxy(c: Composition, aws: str) -> None:
-    for test in tests:
-        mz: Materialized = test["mz"]
-        c.up(mz.name)
-        c.wait_for_materialized(mz.name)
-        c.run("testdrive-svc", aws, *test["td"])
-        c.kill(mz.name)
+def workflow_test(c: Composition, parser: WorkflowArgumentParser) -> None:
```

Comment:
Lets name this `workflow_proxy` to keep up with the general pattern where the main workflow of in `test/XXX` is called `XXX`.

Reply:
Ok, restored. I've been meaning to raise this for a while (and I'll take it to Slack), but I was thinking it'd be good if we could pick a standard name for the main test in in each mzcompose.py. I feel like the first thing I do is figure out what slight deviation of the filename I need in order to get the workflow to run.